packages/js/src/utils/string-utils.ts

```diff
@@ -0,0 +1,153 @@
+/**
```

Comment:
Why is this not moved over to `devkit`?

Doesn't feel like it belongs to `js` package

Reply:
It can be discussed. I was under the impression that Devkit should be for plugin-specific code, whereas JS is a better place for JS-specific logic

Reply:
When you move this to `devkit` make sure to consider there is a util there called `names(name: string): Record<string, string>` we don't want to export 2 ways of creating names.

Reply:
I think it should be combined with https://github.com/nrwl/nx/blob/master/packages/devkit/src/utils/names.ts.

For example, we could use those string functions to replace the arbitrary functions `toClassName`, `toPropertyName`, `toConstantName etc.